deps/rabbit/src/rabbit_vhosts.erl

```diff
@@ -0,0 +1,130 @@
+%% This Source Code Form is subject to the terms of the Mozilla Public
+%% License, v. 2.0. If a copy of the MPL was not distributed with this
+%% file, You can obtain one at https://mozilla.org/MPL/2.0/.
+%%
+%% Copyright (c) 2007-2024 Broadcom. All Rights Reserved. The term “Broadcom” refers to Broadcom Inc. and/or its subsidiaries. All rights reserved.
+%%
+
+%% This module exists to avoid circular module dependencies between
+%% several others virtual hosts-related modules.
+-module(rabbit_vhosts).
+
+-define(PERSISTENT_TERM_COUNTER_KEY, rabbit_vhosts_reconciliation_run_counter).
+
+%% API
+
+-export([
+    list_names/0,
+    exists/1,
+    boot/0,
+    reconcile/0,
+    reconcile_once/0,
+    start_processes_for_all/0,
+    start_on_all_nodes/2,
+    on_node_up/1
+]).
+
+%% Same as rabbit_vhost:exists/1.
+-spec exists(vhost:name()) -> boolean().
+exists(VirtualHost) ->
```

Comment:
Nothing against it, but why this function here? I can't see where it is being used.

Reply:
In fact, `exists/1`, and `list_names/1` already exist in `rabbit_vhost`. Why not use them?

Reply:
Some are in `rabbit_vhost`, others are in the supervision tree-related modules. We need a place for all key functions.

There's also a comment at the top that explains why the module was originally introduced:

```
%% This module exists to avoid circular module dependencies between
%% several others virtual hosts-related modules.
```